vrs/marten-construction-script.py

```diff
@@ -0,0 +1,364 @@
+    # Do collect() and filter() since it is fairly easy just for one field (locus) , instead of an inner join
+    multi_list = group_ht.locus.collect(_localize=False)
+    ht_multiallelic = ht_filt.filter(multi_list.contains(ht_filt.locus))
+    logger.info(
+        f"There are %i very multiallelic sites with %i total variants that will be added",
+        n_multiallelic,
+        ht_multiallelic.count(),
+    )
+
+    # Checkpoint for final multiallelic variants
+    ht_multiallelic = ht_multiallelic.checkpoint(
+        f"{args.tmp_dir}/multiallelic_path_temp.ht",
+        _read_if_exists=args.read_if_exists,
+        overwrite=args.overwrite,
+    )
+
+    # Assemble a table of variants altered by Minimum Representation
+    # NOTE: Construct Table only by partition, assume each partition has ~600 minrep variants
+    ht_minrep = gnomad_v3_genotypes_vds.vds().variant_data.rows()
+    ht_minrep = ht_minrep.annotate(original_alleles=ht_minrep.alleles)
+    ht_minrep = hl.split_multi(ht_minrep)
+    ht_minrep = ht_minrep.filter(
+        ht_minrep.alleles[1] != ht_minrep.original_alleles[ht_minrep.a_index]
+    )
+
+    # 5 partitions to yield 3000 min-rep variants, the default
+    minrep_partitions = 5
+    if args.custom:
+        minrep_partitions = int(counts_list[-1]) // 600
+        if minrep_partitions < 1 and counts_list[-1] != 0:
+            minrep_partitions = 1
+    logger.info(f"Partitions to search for min-rep: %i", minrep_partitions)
+
+    ht_minrep = ht_minrep._filter_partitions(
+        sample(range(ht_minrep.n_partitions()), minrep_partitions)
+    )
+    ht_minrep = ht_minrep.select()
+    ht_minrep = ht_minrep.join(ht_whole, how="inner")
+    # an inner join here joins on Locus and Allele
+    # doing the same using a collect() and filter() step would be awkward, since it involves two separate fields
+
+    # checkpoint for final minrep variants
+    ht_minrep = ht_minrep.checkpoint(
+        f"{args.tmp_dir}/minrep-checkpoint-v2-checking.ht",
+        _read_if_exists=args.read_if_exists,
+        overwrite=args.overwrite,
+    )
+
+    # Counts and further assembly
+    logger.info(f"There are %i variants in the minrep set", ht_minrep.count())
+    ht_final = ht_union.union(ht_multiallelic)
+    ht_final = ht_final.union(ht_minrep)
+
+    # Check for duplicates and de-duplication
+    org_count = ht_final.count()
+    ht_final = ht_final.distinct()
+    duplicates_removed = org_count - ht_final.count()
+    logger.info(f"Number of duplicates removed: %i", duplicates_removed)
+
+    # Export final Table and VCF (both zipped and unzipped) and append header info for missing FILTER descriptions
+    logger.info(f"Before exporting, there are %i total variants", ht_final.count())
+    ht_final = ht_final.checkpoint(
+        f"{args.final_path}/testing-vcf-export.ht",
+        overwrite=args.overwrite,
+        _read_if_exists=args.read_if_exists,
+    )
+    logger.info("VCF as Hail Table written to final path")
```

Comment:
```suggestion
    ht_final = ht_final.checkpoint(
        f"{args.final_path}/testing-vcf-export.ht",
        overwrite=args.overwrite,
        _read_if_exists=args.read_if_exists,
    )
    logger.info("VCF as Hail Table written to final path")
    logger.info(f"Before exporting, there are %i total variants", ht_final.count())
```
(this count will run faster after the checkpoint)